framework/Models/HybridModels/HybridModelBase.py

```diff
@@ -0,0 +1,228 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April, 2020
+
+@author: wangc
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import copy
+import abc
+import numpy as np
+import itertools
+from collections import OrderedDict
+from Decorators.Parallelization import Parallel
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from Models import Dummy
+import Models
+import Files
+from utils import InputData, InputTypes
+#Internal Modules End--------------------------------------------------------------------------------
+
+class HybridModelBase(Dummy):
+  """
+    HybridModel Base Class.
+    A couple of models with the same required inputs and outputs.
+    These models are assembled, and only model is executed to produce the outputs.
+    The selection of models will be determined by the types of hybrid model.
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for the class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying input of cls.
+    """
+    inputSpecification = super(HybridModelBase, cls).getInputSpecification()
+    modelInput = InputData.parameterInputFactory("Model", contentType=InputTypes.StringType)
+    modelInput.addParam("class", InputTypes.StringType)
+    modelInput.addParam("type", InputTypes.StringType)
+    inputSpecification.addSub(modelInput)
+    return inputSpecification
+
+  @classmethod
+  def specializeValidateDict(cls):
+    """
+      This method describes the types of input accepted with a certain role by the model class specialization
+      @ In, None
+      @ Out, None
+    """
+    pass
+
+  def __init__(self,runInfoDict):
+    """
+      Constructor
+      @ In, runInfoDict, dict, the dictionary containing the runInfo (read in the XML input file)
+      @ Out, None
+    """
+    Dummy.__init__(self,runInfoDict)
+    self.modelInstances        = {}                  # dictionary {modelName: modelInstance}: instances of given model
+    self.sleepTime             = 0.005               # waiting time before checking if a run is finished.
+    self.printTag              = 'HybridModelBase MODEL' # print tag
```

Comment:
This doesn't need changing, but note that in the MessageHandler the default trim length for tags is not very long (15 characters as I recall); this might be exceeding that length.